lib/standard/text/abstract_text.nit

```diff
@@ -330,6 +361,46 @@ abstract class Text
 		return if neg then -val else val
 	end
 
+	# Is `self` a valid integer ?
+	#
+	#     assert "0xFE46u8".is_num
+	#     assert "0b0100".is_num
+	#     assert "0o645".is_num
+	#     assert "897u8".is_num
+	fun is_num: Bool do
+		var prefix = get_numhead
+		var s = strip_numhead.strip_numext.remove_all('_')
+		if prefix != "" then
+			var c = prefix[1]
+			if c == 'x' or c == 'X' then return s.is_hex
+			if c == 'o' or c == 'O' then return s.is_oct
+			if c == 'b' or c == 'B' then return s.is_bin
+		end
+		return s.is_dec
+	end
+
+	# Is `self` is a properly formatted integer, returns the corresponding value
+	#
+	#     assert "0xFEu8".to_num  == 254u8
+	#     assert "0b10_10".to_num != 10u8
+	#
+	# REQUIRE: `self.is_num`
+	fun to_num: Numeric do
```

Comment:
This is already inconsistant with to_i since it returns 0 (or something). here to_num aborts